craft_application/commands/lifecycle.py

```diff
@@ -94,7 +100,16 @@ class LifecycleStepCommand(LifecyclePartsCommand):
 
     @override
     def run_managed(self, parsed_args: argparse.Namespace) -> bool:
-        return not parsed_args.destructive_mode
+        """Return whether the command should run in managed mode or not.
+
+        The command will run in managed mode unless the `--destructive-mode` flag
+        is passed OR `CRAFT_BUILD_ENVIRONMENT` is set to `host`.
+        """
+        return _run_managed(parsed_args.destructive_mode)
+
+    @override
+    def provider_name(self, parsed_args: argparse.Namespace) -> str | None:
```

Comment:
This can also go into `LifecyclePartsCommand` rather than being here and in `CleanCommand`